File: app/src/main/java/com/photoeditor/sketch/photo/maker/editor/sketchphotoeditor/pencilsketching/sketchphoto/Ragnarok/MotionBlurFilter.java
package com.photoeditor.sketch.photo.maker.editor.sketchphotoeditor.pencilsketching.sketchphoto.Ragnarok;

import android.graphics.Bitmap;

class MotionBlurFilter {

	static {
		System.loadLibrary("AndroidImageFilter");
	}
	
	static Bitmap changeToMotionBlur(Bitmap bitmap, int xSpeed, int ySpeed) {
		int width = bitmap.getWidth();
		int height = bitmap.getHeight();
		
		int[] pixels = new int[width * height];
		bitmap.getPixels(pixels, 0, width, 0, 0, width, height);
		
		int[] returnPixels = NativeFilterFunc.motionBlurFilter(pixels, width, height, xSpeed, ySpeed);
		return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888);
	}
}

File: app/src/main/java/com/sketch/ragnarok/MotionBlurFilter.kt
package com.sketch.ragnarok

import android.graphics.Bitmap

internal object MotionBlurFilter {

    init {
        System.loadLibrary("AndroidImageFilter")
    }

    fun changeToMotionBlur(bitmap: Bitmap, xSpeed: Int, ySpeed: Int): Bitmap {
        val width = bitmap.width
        val height = bitmap.height
        val pixels = IntArray(width * height)
        bitmap.getPixels(pixels, 0, width, 0, 0, width, height)
        val returnPixels = NativeFilterFunc.motionBlurFilter(pixels, width, height, xSpeed, ySpeed)
        return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888)
    }
}